test/built-ins/BigInt/asIntN/bits-toindex.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2017 Josh Wolfe. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: BigInt.asIntN type coercion for bits parameter
+info: >
+  BigInt.asIntN ( bits, bigint )
+
+  1. Let bits be ? ToIndex(bits).
+features: [Symbol]
+---*/
+
+function MyError() {}
+
+assert.sameValue(BigInt.asIntN(undefined, 1n), 0n);
+assert.sameValue(BigInt.asIntN(null, 1n), 0n);
+assert.sameValue(BigInt.asIntN(true, 1n), -1n);
+assert.sameValue(BigInt.asIntN(false, 1n), 0n);
+assert.sameValue(BigInt.asIntN("foo", 1n), 0n);
+assert.sameValue(BigInt.asIntN("3", 10n), 2n);
+assert.throws(TypeError, () => BigInt.asIntN(Symbol(0), 0n));
+assert.throws(TypeError, () => BigInt.asIntN(1n, 0n));
+assert.sameValue(BigInt.asIntN(Object(3), 10n), 2n);
+assert.sameValue(BigInt.asIntN({valueOf:()=>3}, 10n), 2n);
```

Comment:
I can add tests with `Symbol.toPrimitive` methods. I originally shied away from writing thorough tests for ToPrimitive for such a distant method as `BigInt.asIntN`, but if that's what we want, I can do that.

I figured at some point implementations are going to be using the same ToPrimitive implementation that they got test coverage for in another context. It seems like more work to violate the details of ToPrimitive for only BigInt.asIntN than to just use a common ToPrimitive implementation. So it seems not very important to test every corner case of every reachable abstract operation from every user-facing API, but I can see the appeal of that approach as well.

Should I throw in some `Symbol.toPrimitive` tests? If so, I think I agree that they should be sectioned out into a separate test file.